### PR TITLE
refactoring postgresql controller

### PR DESCRIPTION
**Pull request by mweibel**
## Summary

Includes the changes agreed on in #50. 

Note: this is based upon #52 because of the version compare changes, among other things. 

## Checklist

- [x] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog
- [x] Update tests.
- [x] Link this PR to related issues.
- [x] I have run `make test-e2e` and e2e tests pass successfully

### CODE REVIEW

File: operator/postgresqlcontroller/observe.go [hunk 1]
```diff
@@ -64,3 +105,81 @@ func setConditionFromState(pgExo oapi.DbaasServicePg, pgInstance *exoscalev1.Pos
...
+		if !v {
+			log.V(2).Info("instance not up-to-date", "check", v)
+			ok = false
+		}
+	}
+	return ok
+}
+
+// toConnectionDetails parses the connection details from the given observation.
+func toConnectionDetails(pgExo oapi.DbaasServicePg, ca string) (managed.ConnectionDetails, error) {
```
Comment (susana-garcia) [inline operator/postgresqlcontroller/observe.go:169]: nit: maybe rename it to `connectionDetails` to be consistent with other resource's observe methods

**Comment by susana-garcia** [comment]
LGTM

**Review by susana-garcia — APPROVED** [review]

**Review by ZuGao — COMMENTED** [review]
Just small things to correct. I ran e2e tests and one test failed for mysql... Can you check try to correct it?

**Comment by mweibel** [comment]
> LGTM, but maybe (git) drop the first commit as it's already in master? Also not 100% sure is needed thinking

nice catch - didn't notice that before. Might've been a rebase gone wrong. Fixed now.

**Review by ZuGao — APPROVED** [review]
LGTM

**Pull request merged**